.github/workflows/compliance.yml

```
@@ -0,0 +1,140 @@
+#####################################################################################################################
+# @author       Michael Smith
+# @date         2021-08-23
+# @description  As SFDO Managed Packages start using the newly introduced Instrumentation Services to 'log' 
+#               usage data to Splunk and Argus, this Compliance Check action does two things:
+#               1. Ensures that Packaged Apex Classes do not make any direct reference to the core SfdoLogUtils class.
+#                  Any code references to this class will cause the Compliance build to fail.
+#               2. At least in the short term, adds an Instrumentation reviewer to any PR that in any way references or
+#                  or uses the SfdoInstrumentationService. This is primarily meant as a short term action to aid in
+#                  adoption of the new instrumentation services.
+#####################################################################################################################
+name: "Instrumentation Compliance"
+on:
+  pull_request:
+    # Filter the job to only execute if CLS files (not in the unpackaged folder) are in the PR
+    paths:    
+      - '**.cls'
+      - '!unpackaged/**.cls'
+
+jobs:
+  Instrumentation_Service_Compliance_Verification:
+
+    # Constant vars
+    # These should be adjusted per repository
+    env:
+      DEFAULT_REVIEWER: force2b
+      ALTERNATE_REVIEWER: lparrott
+      ASSIGNED_LABEL: Instrumentation
+
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Get PR Owner
+        id: pr_owner
+        run: |
+          echo "::set-output name=owner::${{ github.event.pull_request.user.login }}"
+
+      - name: Set Reviewer
+        id: set_reviewer
+        run: |
+          if [${{ steps.pr_owner.outputs.owner != env.DEFAULT_REVIEWER}}]
```

This won't quite work, since it'll be treated as a command. I suggest:
```suggestion
          if [[ "${{ steps.pr_owner.outputs.owner != env.DEFAULT_REVIEWER}}" == "true" ]]
```

(Comparing `owner` and env var directly can also work.)

---

@jstvz  I think this was working though. There were points where I tested changes and Luke was assigned as the reviewer when I was the PR owner.

---

Ah, I see the problem. I'm seeing this in the [log](https://github.com/SalesforceFoundation/NPSP/runs/4216032180?check_suite_focus=true#step:4:13):
```
/home/runner/work/_temp/5b944cf8-50e2-4dfe-b4e3-12a048928657.sh: line 1: [false]: command not found
```

Because the test-command returns a non-zero exit code when `command not found`, the so the else clause is always executed. For example:
```console
bash-3.2$ if [false]; then echo 'false'; else echo 'true';fi
bash: [false]: command not found
true
```

---

Thanks. Just to confirm - the line does fail, but it still works; which means I should make that change?

---

Yes. This is a "stopped clock"; you could transpose the values for default and alternate reviewers and the alternate would still be selected.